src/TorchSharp/DataLoader.cs

```diff
@@ -110,18 +110,25 @@ private bool MoveNextValue()
                         public bool MoveNext()
                         {
                             DisposeCurrent();
-                            if (!MoveNextValue()) return false;
-                            List<Dictionary<string, Tensor>> dic = new();
-                            dic.Add(dataset.GetTensor(currentVal));
-                            for (var i = 1; i < batchSize; i++) {
-                                if (!MoveNextValue()) break;
+                            using (var scope = DisposeScopeManager.NewDisposeScope()) {
```

Comment:
NIT: using var scope = ...;

Reply:
Yeah, I know...

I do usually end up relying on the `using ( ... ) { ... }` syntax in most circumstances. Maybe it's because it was what I got used to when 'using' first was introduced.